### PR TITLE
Switch to SQLite3, Upgrade Ruby and Rails, Remove Unused Gems, and Add Fly.io Deployment Configuration

### PR DESCRIPTION
**Pull request by cirdes**
New server: https://rbui.fly.dev

This PR includes the following updates and changes:

1. **Database Switch**:
   - Removed PostgreSQL and Redis, and switched to SQLite3 for simpler web stack.

2. **Version Upgrades**:
   - Updated Ruby to version 3.3.4.
   - Updated Rails to version 7.2.0.beta3.

3. **Asset Pipeline Update**:
   - Removed Sprockets and added Propshaft for managing assets.

4. **Removed Unused Gems**:
   - Removed the following unused gems:
     - `stripe`
     - `letter_opener` (development group)
     - `postmark-rails` (used in production)
     - `octokit` (version ~> 5.0)
     - `faraday-retry` (version ~> 2.0) - used for API calls failing in the `octokit` gem
     - `rack-www` - used to redirect www to non-www subdomain
     - `rollbar`

5. **Deployment Configuration**:
   - Added configuration for deployment on Fly.io.

These changes streamline the project dependencies, simplify development, and prepare the application for deployment on Fly.io.

### CODE REVIEW

File: .yarn/releases/yarn-1.22.19.cjs
Comment (iseth) [inline .yarn/releases/yarn-1.22.19.cjs:1]: can we get rid of this?

File: .yarnrc
Comment (iseth) [inline .yarnrc:1]: can we get rid of this?

File: Gemfile [hunk 1]
```diff
@@ -1,37 +1,30 @@
...
-gem "stimulus-rails"
-
+gem "stimulus-rails", "1.3.3"
 # Bundle and process CSS [https://github.com/rails/cssbundling-rails]
-gem "cssbundling-rails"
+gem "cssbundling-rails", "1.4.0"
 
-# Build JSON APIs with ease [https://github.com/rails/jbuilder]
-gem "jbuilder"
+# # Build JSON APIs with ease [https://github.com/rails/jbuilder]
```
Comment (iseth) [inline Gemfile:23]: this too haha

File: package.json [hunk 2]
```diff
@@ -2,21 +2,22 @@
...
+    "tailwindcss-animate": "1.0.7"
   },
   "scripts": {
-    "build": "esbuild app/javascript/*.* --bundle --sourcemap --outdir=app/assets/builds --public-path=/assets",
+    "build": "esbuild app/javascript/*.* --bundle --sourcemap --format=esm --outdir=app/assets/builds --public-path=/assets",
     "build:css": "tailwindcss -i ./app/assets/stylesheets/application.tailwind.css -o ./app/assets/builds/application.css --minify"
-  }
-}
+  },
+  "packageManager": "yarn@1.22.19"
```
Comment (iseth) [inline package.json:22]: this might have something to do with the yarn release files. Look into removing this.

**Review by iseth — CHANGES_REQUESTED** [review]
almost perfect

File: .yarn/releases/yarn-1.22.19.cjs
Comment (cirdes) [reply .yarn/releases/yarn-1.22.19.cjs:1]: @iseth, I have tested running `fly launch` from the `main` branch here, and it creates all of those files. Maybe we should keep it. 

Screenshot 2024-07-29 at 11 43 19 AM

File: package.json (hunk 2 above)
Comment (cirdes) [reply package.json:22]: This was also added by the Fly CLI. It seems to be a good practice to specify the Yarn version here to ensure that the specific Yarn version is always used.

Screenshot 2024-07-29 at 11 51 37 AM

File: .yarnrc
Comment (cirdes) [reply .yarnrc:1]: I don't think we should because it was generated by fly

Comment (iseth) [reply .yarnrc:1]: Yeah i just got things working on my end and saw that too. Should it really be checked in to git though?

**Comment by cirdes** [comment]
@iseth, I have removed the ".tool-versions" file and fixed the comment in the Gemfile.

All other issues you mentioned were generated by the fly.io CLI, and I believe they are recommended by Yarn.

Let me know if you need anything else!

**Comment by iseth** [comment]
> @iseth, I have removed the ".tool-versions" file and fixed the comment in the Gemfile.
> 
> All other issues you mentioned were generated by the fly.io CLI, and I believe they are recommended by Yarn.
> 
> Let me know if you need anything else!

I dig too deep into this but i think its definetly coming from https://github.com/fly-apps/dockerfile-rails/blob/2711cf5aab3716c3dafd1c08307ee62107a45dda/lib/generators/templates/_npm_install.erb

**Comment by cirdes** [comment]
@iseth, I have removed .yarn/* and .yarnrc and added them to .gitignore.

**Review by iseth — APPROVED** [review]
awesome!

**Pull request merged**